### PR TITLE
feat: add aws sns context propagation and DSM tests

### PR DESCRIPTION
**Pull request by wconti27**
## Motivation

More testing

## Changes

Adds AWS SNS Context Propagation Test with SQS as a consumer and DSM Tests. Also adds test application for nodejs and python

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
    * [ ] To R&P team: locally build and push the image to hub.docker.com 
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
    * [ ] Once merged, add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: tests/integrations/crossed_integrations/test_sns_to_sqs.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+#     @missing_feature(library="nodejs", reason="Expected to fail, Nodejs does not propagate context")
+#     def test_produce_trace_equality(self):
+#         super().test_produce_trace_equality()
+
+#     @missing_feature(library="golang", reason="Expected to fail, Golang does not propagate context")
+#     @missing_feature(library="ruby", reason="Expected to fail, Ruby does not propagate context")
+#     @missing_feature(library="python", reason="Expected to fail, Python does not propagate context")
+#     @missing_feature(library="nodejs", reason="Expected to fail, Nodejs does not propagate context")
+#     def test_consume_trace_equality(self):
+#         super().test_consume_trace_equality()
```
Comment (robertomonteromiguel) [inline tests/integrations/crossed_integrations/test_sns_to_sqs.py:261]: can we remove this?

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
Minor suggested changes.
Revert the ci.yml change!

File: tests/integrations/crossed_integrations/test_sns_to_sqs.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    @staticmethod
+    def get_topic(span) -> str | None:
+        """Extracts the queue from a span by trying various fields"""
+        topic = span["meta"].get("topicname", None)  # this is in nodejs, java, python
+
+        if topic is None:
+            if "aws.sns.topic_arn" in span["meta"]:
+                topic = span["meta"]["aws.sns.topic_arn"].split(":")[-1]
+            # elif "messaging.url" in span["meta"]:
+            #     topic = span["meta"]["messaging.url"].split("/")[-1]
```
[the remark this answers is not in the document]
Comment (wconti27) [reply tests/integrations/crossed_integrations/test_sns_to_sqs.py:98]: Done

File: tests/integrations/crossed_integrations/test_sns_to_sqs.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+                queue = span["meta"]["messaging.url"].split("/")[-1]
+
+            if queue is None:
+                logger.error(f"could not extract queue from this span:\n{span}")
+
+        return queue
+
+    @staticmethod
+    def get_topic(span) -> str | None:
+        """Extracts the queue from a span by trying various fields"""
```
[the remark this answers is not in the document]
Comment (wconti27) [reply tests/integrations/crossed_integrations/test_sns_to_sqs.py:91]: Fixed

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -37,7 +37,7 @@ jobs:
     strategy:
       matrix:
         os:
-          - macos-latest
+          # - macos-latest
```
[the remark this answers is not in the document]
Comment (wconti27) [reply .github/workflows/ci.yml:40]: Done, sometimes the mac-os tests take upwards of 30 minutes to get a runner and actually run! Much faster to comment it out, then restore later.

File: tests/integrations/crossed_integrations/test_sns_to_sqs.py (hunk 1 above)
Comment (wconti27) [reply tests/integrations/crossed_integrations/test_sns_to_sqs.py:282]: done

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**